OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -36,12 +36,15 @@ public FlyAttack(Actor self, Target target)
 		public override Activity Tick(Actor self)
 		{
 			if (!target.IsValidFor(self))
+			{
+				self.QueueActivity(new ReturnToBase(self));
 				return NextActivity;
+			}
 
 			// Move to the next activity only if all ammo pools are depleted and none reload automatically
 			// TODO: This should check whether there is ammo left that is actually suitable for the target
 			if (ammoPools.All(x => !x.Info.SelfReloads && !x.HasAmmo()))
-				return NextActivity;
+				return new ReturnToBase(self);
```

Comment:
This is going to erase the order queue.  You need to sequence the RTB before continuing the NextActivity with something like `return ActivityUtils.SequenceActivities(new ReturnToBase(self), NextActivity)`.